src/sync.d

```diff
@@ -4591,10 +4591,17 @@ class SyncEngine {
 							
 							// No 404 or otherwise was triggered, meaning that the file already exists online and passes the POSIX test ...
 							log.vdebug("fileDetailsFromOneDrive after exist online check: ", fileDetailsFromOneDrive);
+
+							// Save item to the database
+							saveItem(fileDetailsFromOneDrive);
+							
```

Comment:
I do not think the updated logic here is correct.

If the response from `fileDetailsFromOneDrive = checkFileOneDriveApiInstance.getPathDetailsByDriveId(parentItem.driveId, fileToUpload)` generates a 200 we drop to the POSIX test.

In your code, you are then saving this data blindly to the DB, without validating that the details from online match what was intended to upload - so that DB save should not be happening until the validation of the data is done

Please can you review your thinking here

Reply:
My idea was that if we maintain the latest online version in the local database, it can help force triggering a reupload 
 later during the database inconsistency check in case the corresponding upload process fail. Otherwise, there might be a risk of losing this inconsistency.